apps/etterlatte-grunnlag/src/main/kotlin/grunnlag/GrunnlagRoutes.kt

```diff
@@ -82,21 +83,27 @@ fun Route.grunnlagRoute(grunnlagService: GrunnlagService, behandlingKlient: Beha
                 HttpStatusCode.Unauthorized,
                 "Kunne ikke hente ut navident for vurdering av ytelsen kommer barnet tilgode"
             )
-
-            withFoedselsnummer(behandlingKlient) { foedselsnummer ->
-                val opplysning = grunnlagService.hentOpplysningstypeNavnFraFnr(
-                    foedselsnummer,
-                    navIdent
-                )
-
-                if (opplysning != null) {
-                    call.respond(opplysning)
-                } else {
-                    call.respond(
-                        HttpStatusCode.NotFound,
-                        "Gjenny har ingen navnedata på fødselsnummeret som ble etterspurt"
+            try {
+                withFoedselsnummer(behandlingKlient) { foedselsnummer ->
+                    val opplysning = grunnlagService.hentOpplysningstypeNavnFraFnr(
+                        foedselsnummer,
+                        navIdent
                     )
+
+                    if (opplysning != null) {
+                        call.respond(opplysning)
+                    } else {
+                        call.respond(
+                            HttpStatusCode.NotFound,
+                            "Gjenny har ingen navnedata på fødselsnummeret som ble etterspurt"
+                        )
+                    }
                 }
+            } catch (ex: InvalidFoedselsnummerException) {
```

Comment:
Hvorfor ikke svare med en `400 Bad request` her?

Reply:
Fordi vi vil ikke avsløre noe til saksbehandler, standard her er 500

Reply:
Avsløre hva da? At fødselsnummeret ikke passerer en strukturell sjekk og sannsynligvis er skrevet feil? Det må være grenser for hva vi beskytter oss i mot, og dette gjør det bare umulig å hjelpe saksbehandler i dette tilfellet. Jeg kan være enig på ting annet enn InvalidFoedselsnummerException er det ok å maskere det, men her vet vi at vi ikke snakker om et reelt foedselsnummer en gang

Reply:
https://github.com/navikt/pensjon-etterlatte-saksbehandling/pull/1557/commits/0e40b167be4112dc19b059b8703144b913062ef4

Reply:
Enig, my bad